webclient/webclient/src/main/java/io/helidon/webclient/NettyClientHandler.java

```diff
@@ -236,6 +226,17 @@ protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws IO
         }
     }
 
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        super.channelInactive(ctx);
+        // Connection closed without last HTTP content received. Some server problem
+        // so we need to fail the publisher and report an exception.
+        if (!responseCloser.isClosed()) {
+            WebClientException exception = new WebClientException("Connection reset by the host");
+            publisher.fail(exception);
+        }
```

Comment:
There is null check in the `channelReadComplete` I would expect same is needed here, previous version captured it from the `channelRead0` context.
```suggestion
        if (publisher != null && !responseCloser.isClosed()) {
            WebClientException exception = new WebClientException("Connection reset by the host");
            publisher.fail(exception);
        }
```